### PR TITLE
Add functions to geo module

### PR DESCRIPTION
**Pull request by pabloarosado**

### CODE REVIEW

File: owid/datautils/geo.py [hunk 1]
```diff
@@ -1,3 +1,491 @@
...
+# Maximum number of nans allowed per year when doing aggregations (None to allow any number of nans).
+NUM_ALLOWED_NANS_PER_YEAR = None
+
+
+def _load_population() -> pd.DataFrame:
+    population = (
+        catalog.find("population", namespace="owid", dataset="key_indicators")
+        .load()
+        .reset_index()
+    )  # type: pd.DataFrame
```
Comment (Marigold) [inline owid/datautils/geo.py:36]: Nitpick - AFAIK type annotations should be preferred over type comments. You can write this as
```
population: pd.DataFrame = (
        catalog.find("population", namespace="owid", dataset="key_indicators")
        .load()
        .reset_index()
    )
```
or 
```
return cast(pd.DataFrame, population)
```

File: owid/datautils/geo.py [hunk 2]
```diff
@@ -1,3 +1,491 @@
...
+        income_groups_found[income_groups_found.path.str.startswith("garden")]
+        .load()
+        .reset_index()
+    )  # type: pd.DataFrame
+
+    return income_groups
+
+
+def _warn_on_list_of_entities(
+    list_of_entities: Union[List[str], Set[str]], warning_message: str, show_list: bool
```
Comment (Marigold) [inline owid/datautils/geo.py:63]: You could use `Sequence[str]` here as well. Up to you.

**Review by Marigold — APPROVED** [review]
Looks good! We already have some util functions for regions & population in `etl`, if you stumble upon them, don't be afraid to replace them with your version.

**Comment by pabloarosado** [comment]
> Looks good! We already have some util functions for regions & population in `etl`, if you stumble upon them, don't be afraid to replace them with your version.

Thanks @Marigold, I'd say that once these functions are fully featured and we agree on their functionality, `etl` should import them from datautils, instead of having their own functions.

File: owid/datautils/geo.py (hunk 2 above)
Comment (pabloarosado) [reply owid/datautils/geo.py:63]: Are you sure? When I use `list_of_entities: Sequence[str]`, then `list_of_entities` cannot be a `set` (I get an error about [line 480](https://github.com/owid/data-utils-py/blob/2ac06575a302a57b31cd8e88881bc89b312d7847/owid/datautils/geo.py#L480): `Argument "list_of_entities" to "_warn_on_list_of_entities" has incompatible type "Set[Any]"; expected "Sequence[str]"`).

File: owid/datautils/geo.py (hunk 1 above)
Comment (pabloarosado) [reply owid/datautils/geo.py]: Thanks, done.

File: owid/datautils/geo.py (hunk 2 above)
Comment (Marigold) [reply owid/datautils/geo.py:63]: Sorry, you're right. I've mistakenly read it as `List` and `Tuple` 🤦

**Comment by pabloarosado** [comment]
Hi @Marigold , a few comments on this old PR:
1. I just realised I was using `assert are_equal(...)` in the tests. But `are_equal()` returns a tuple of a boolean and a dataframe, therefore the assertions were always True. After fixing this, I realised some tests were not passing (because I was not properly sorting some of the outputs).
2. This makes me think that it would be good if `are_equal` just returns a boolean, to avoid confusion. Maybe we could have a class `Comparison`, with the methods `are_equal`, and `differences`, for example. But I'd suggest to do this in a different release.
3. Apart from that, I added a new feature to `add_region_aggregates`: optionally keep data for original regions, adding a suffix to its name.

@lucasrodes do you think you can find some time to check this PR?

**Pull request merged**